### PR TITLE
DataGrid - add export to pdf button

### PR DESCRIPTION
**Pull request by pomahtri**
[jquery](https://gist.github.com/pomahtri/8b46f6afaa02cb7458e26485870ffc19)

todo:

- [x] fix localization
- [x] remove `fileName`
- [ ] add separator between excel and pdf (will implement in next PR)

@DevExpress/devextreme-navigation, also please check:

- [x] Props passed to `onExporting` callback
  - [x] `selectedRowsOnly`
  - [x] `format`

@AnnaKondratova, also please check:

- [ ] new option `export.formats` with type `('xlsx' | 'pdf' | string)[]` and default value `[ 'xlsx' ]`
- [ ] new arguments passed to `onExporting` callback
  - `selectedRowsOnly`
  - `format`
- [ ] usage of "xlsx" instead of "excel"

### CODE REVIEW

**Review by AnnaKondratova — DISMISSED** [review]

**Review by AnnaKondratova — DISMISSED** [review]

File: js/ui/data_grid/ui.data_grid.export.js [hunk 1]
```diff
@@ -821,7 +862,7 @@ dataGridCore.registerModule('export', {
 
                 optionChanged: function(args) {
                     this.callBase(args);
-                    if(args.name === 'export') {
+                    if(args.name === 'export' || args.name === 'pdfExport') {
```
Comment (NickMitrokhin) [inline js/ui/data_grid/ui.data_grid.export.js]: what is pdfExport for?

File: testing/tests/DevExpress.ui.widgets.dataGrid/exportController.tests.js [hunk 2]
```diff
@@ -2314,7 +2355,10 @@ QUnit.module('Export menu', {
     QUnit.test('The export button is not shown', function(assert) {
         this.setupModules({
             'export': {
-                enabled: false
+                enabled: false,
+            },
+            'pdfExport': {
```
Comment (NickMitrokhin) [inline testing/tests/DevExpress.ui.widgets.dataGrid/exportController.tests.js]: what is pdfExport for?

**Review by NickMitrokhin — CHANGES_REQUESTED** [review]

**Review by NickMitrokhin — DISMISSED** [review]

File: js/ui/data_grid.d.ts [hunk 3]
```diff
@@ -2867,7 +2867,9 @@ export type ExportedEvent<TRowData = any, TKey = any> = EventInfo<dxDataGrid<TRo
 
 /** @public */
 export type ExportingEvent<TRowData = any, TKey = any> = Cancelable & EventInfo<dxDataGrid<TRowData, TKey>> & {
-  fileName?: string;
+  fileName: string;
```
Comment (EugeniyKiyashko) [inline js/ui/data_grid.d.ts]: fileName?: string

File: ts/dx.all.d.ts [hunk 4]
```diff
@@ -7221,7 +7225,9 @@ declare module DevExpress.ui {
       TKey = any
     > = DevExpress.events.Cancelable &
       DevExpress.events.EventInfo<dxDataGrid<TRowData, TKey>> & {
-        fileName?: string;
+        fileName: string;
```
Comment (EugeniyKiyashko) [inline ts/dx.all.d.ts:7228]: fileName?: string;

File: js/ui/data_grid/ui.data_grid.export.js [hunk 5]
```diff
@@ -761,24 +779,49 @@ dataGridCore.registerModule('export', {
...
-                            }
+                            },
                         });
 
                         if(exportOptions.allowExportSelectedData) {
                             items.push({
-                                text: texts.exportSelectedRows,
-                                icon: DATAGRID_EXPORT_SELECTED_ICON,
+                                text: format(texts.exportSelectedRows, 'Excel'),
+                                icon: pdfEnabled ? DATAGRID_EXPORT_EXCEL_ICON : DATAGRID_EXPORT_SELECTED_ICON,
```
Comment (EugeniyKiyashko) [inline js/ui/data_grid/ui.data_grid.export.js]: Is it typo? Checking pdfEnabled in excel section? 
icon: pdfEnabled ? DATAGRID_EXPORT_EXCEL_ICON : DATAGRID_EXPORT_SELECTED_ICON,

Comment (pomahtri) [reply js/ui/data_grid/ui.data_grid.export.js]: It was by design
Now changed because of different design of icon for "export selected to pdf"

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by NickMitrokhin — APPROVED** [review]

**Review by Alyar666 — APPROVED** [review]

**Pull request merged**